services/delegation/src/modules/delegation/delegation.service.ts

```diff
@@ -590,4 +604,69 @@ export class DelegationService {
       .getRawOne();
     return parseInt(d.total, 10) || 0;
   }
+
+  private async undelegateFromChain(
+    nodeId: number,
+    type: string,
+    address: string,
+    amount: BigNumber,
+  ) {
+    let undelegationBlockNumber;
+    const targetChain =
```

Comment:
Don't see the point of this.

Reply:
We need a check later on, but I need to add this ti the ifs too so that it makes sense, yes.